pkg/controller/kubelet-config/helpers.go

```diff
@@ -92,7 +92,7 @@ func getManagedKubeletConfigKeyDeprecated(pool *mcfgv1.MachineConfigPool) string
 
 // validates a KubeletConfig and returns an error if invalid
 func validateUserKubeletConfig(cfg *mcfgv1.KubeletConfig) error {
-	if cfg.Spec.KubeletConfig.Raw == nil {
+	if cfg.Spec.KubeletConfig == nil || cfg.Spec.KubeletConfig.Raw == nil {
```

Comment:
Curious, do we consider Spec.KubeletConfig mandatory at all? I assume no since e.g. a user should be able to create a kubeletconfig to just modify the logLevel via https://github.com/openshift/machine-config-operator/blob/master/install/0000_80_machine-config-operator_01_kubeletconfig.crd.yaml#L144 ?

Reply:
@yuqi-zhang  No, it's not mandatory so the [Line96](https://github.com/openshift/machine-config-operator/pull/2699/files#diff-4a2d0a4442e10d889a927322ad7311c67b22b259622067b52cbda7d2ece3ebefR96) does not rise an error. But if `Spec.KubeletConfig` is nil the `cfg.Spec.KubeletConfig.Raw` will lead to nil pointer panic.